runtime/domain.c

```diff
@@ -1106,6 +1115,60 @@ int caml_domain_is_in_stw(void) {
 }
 #endif
 
+/* During a stop-the-world (STW), all currently running domains stop
+   their usual work and synchronize to all call the same function.
+
+   STW sections use [all_domains_lock] and the variable [stw_leader]
+   (0 when no STW section is running, the dom_internal* pointer of the
+   STW leader when a STW section is running) to guarantee that no
+   domain is running something else:
+
+   - If two STW sections are attempted in parallel, only one will
+     manage to take the lock, and the domain starting the other will
+     join that winning STW section, without running its own STW code
+     at all. (This is the [_try_] in the function name: if it returns
+     0, the STW section did not run at all, so you should call this
+     function in a loop.)
+
+   - Domain initialization code from [create_domain] will not run in
+     parallel with a STW section, as [create_domain] starts by
+     looping until (1) it has the [all_domains_lock] and (2) there is
+     no current STW section (using the [stw_leader] variable).
+
+   - Domain cleanup code runs after the terminating domain may run in
+     parallel to a STW section, but only after that domain has safely
+     removed itself from the STW participant set: the
+     [domain_terminate] function is careful to only leave the STW set
+     when (1) it has the [all_domains_lock] and (2) it hasn't received
+     any request to participate in a STW section.
+
+   Each domain leaves the section as soon as it is finished running
+   the section code. In particular, a mutator may resume while some
+   other domain are still in the section. Section code that needs to
```

Comment:
"Section code" -> "The section of code"

Reply:
I meant "the code of the STW section", which I think is different. I went over the whole documentation to talk more explicitly about a "STW callback" (the function that all participants are running), and mention "section callback" or "STW callback" instead of the more vague "section code". THanks!